nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -34,6 +34,7 @@
 import java.io.InputStream;
 import java.io.Reader;
 import java.lang.reflect.Array;
+import java.math.BigDecimal;
```

Comment:
Currently
```java
DataTypeUtils.findMostSuitableType(
        new BigDecimal("11.110"),
        Arrays.asList(RecordFieldType.DECIMAL.getDecimalDataType(5, 3)),
        Function.identity()
);
```
correctly returns `Optional.of(RecordFieldType.DECIMAL.getDecimalDataType(5, 3));`, but
```java
DataTypeUtils.findMostSuitableType(
        new BigDecimal("1.110"),
        Arrays.asList(RecordFieldType.DECIMAL.getDecimalDataType(5, 3)),
        Function.identity()
);
```
("1.110" instead of "11.111") returns `Optional.empty();`

This is because the inferred type DECIMAL(4,2) is not equal to the provided DECIMAL(5,2) so the `getWiderType` method is called which doesn't handle cases where the two types are the same in general, only differ in their details.

Probably the `public static Optional<DataType> getWiderType(final DataType thisDataType, final DataType otherDataType)` should handle two different DECIMAL types by returning another DECIMAL that is the superset of the two in terms of scale and precision.